### PR TITLE
feat(consensus): add trust metric for consensus

### PR DESCRIPTION
**Pull request by KaoImin**
**What type of PR is this?**
feature

**What this PR does / why we need it**:
* add trust metric for consensus

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

### CODE REVIEW

File: core/consensus/src/engine.rs [hunk 1]
```diff
@@ -413,6 +414,17 @@ impl<Adapter: ConsensusAdapter + 'static> Wal for ConsensusEngine<Adapter> {
     }
 }
 
+impl<Adapter: ConsensusAdapter + 'static> Report for ConsensusEngine<Adapter> {
+    fn report_error(&self, ctx: Context, err: OverlordError) {
+        match err {
+            OverlordError::CryptoErr(_) | OverlordError::AggregatedSignatureErr(_) => self
+                .adapter
+                .report_bad(ctx, TrustFeedback::Bad(err.to_string())),
```
Comment (zeroqn) [inline core/consensus/src/engine.rs:422]: Bad? Not worse?

File: core/consensus/src/message.rs [hunk 2]
```diff
@@ -85,10 +85,13 @@ impl<C: Consensus + 'static> ProposalMessageHandler<C> {
 impl<C: Consensus + 'static> MessageHandler for ProposalMessageHandler<C> {
     type Message = Proposal;
 
-    async fn process(&self, ctx: Context, msg: Self::Message) {
+    async fn process(&self, ctx: Context, msg: Self::Message) -> TrustFeedback {
         if let Err(e) = self.consensus.set_proposal(ctx, msg.0).await {
```
Comment (zeroqn) [inline core/consensus/src/message.rs:89]: Should we match err here? exclude consensus internal error?

Comment (zeroqn) [inline core/consensus/src/message.rs:89]: Does this exclude consensus internal error?

File: core/consensus/src/message.rs [hunk 3]
```diff
@@ -85,10 +85,13 @@ impl<C: Consensus + 'static> ProposalMessageHandler<C> {
...
             warn!("set proposal {}", e);
+            return TrustFeedback::Bad(e.to_string());
         }
+
+        TrustFeedback::Neutral
```
Comment (zeroqn) [inline core/consensus/src/message.rs:94]: IMO, this should report TrustFeedback::Good.

File: core/consensus/src/message.rs (hunk 2 above)
Comment (KaoImin) [reply core/consensus/src/message.rs:89]: Yes, the error can be only caused by `DecodeErr`.

**Comment by zeroqn** [comment]
Close, will make another pr.

**Pull request closed without merging**